src/otx/v2/adapters/torch/dataset.py

```diff
@@ -0,0 +1,210 @@
+class TorchBaseDataset(BaseDataset):
+    """A class representing a dataset for pretraining a model."""
+
+    def _initialize(self) -> None:
+        self.set_datumaro_adapters()  # Set self.dataset_entity & self.label_schema
+        self.initialize = True
+
+    def _build_dataset(
+        self,
+        subset: str,
+        pipeline: list | dict | None = None,
+        config: str | dict | None = None,
+    ) -> TorchDataset | None:
+        """Builds a TorchDataset object for the given subset using the specified pipeline and configuration.
+
+        Args:
+            subset (str): The subset to build the dataset for.
+            pipeline (Optional[Union[list, dict]]): The pipeline to use for the dataset.
+                Defaults to None.
+            config (Optional[Union[str, dict]]): The configuration to use for the dataset.
+                Defaults to None.
+
+        Returns:
+            Optional[TorchDataset]: The built TorchDataset object, or None if the dataset is empty.
+        """
+        raise NotImplementedError
+
+
+    def _build_dataloader(
+        self,
+        dataset: TorchDataset | None,
+        batch_size: int | None = 2,
+        num_workers: int | None = 0,
+        shuffle: bool = True,
+        pin_memory: bool = False,
+        drop_last: bool = False,
+        sampler: Sampler | (Iterable | dict) | None = None,
+        persistent_workers: bool = False,
+        **kwargs,
+    ) -> TorchDataLoader | None:
+        """Builds a PyTorch DataLoader for the given dataset.
+
+        Args:
+            dataset (Optional[TorchDataset]): The dataset to load.
+            batch_size (int): The batch size to use.
+            num_workers (int): The number of worker processes to use for data loading.
+            shuffle (bool): Whether to shuffle the data.
+            pin_memory (bool): Whether to pin memory for faster GPU transfer.
+            drop_last (bool): Whether to drop the last incomplete batch.
+            sampler (Optional[Union[Sampler, Iterable, Dict]]): The sampler to use for data loading.
+            persistent_workers (bool): Whether to keep the worker processes alive between iterations.
+            **kwargs: Additional arguments to pass to the DataLoader constructor.
+
+        Returns:
+            Optional[TorchDataLoader]: The DataLoader for the given dataset.
+        """
+        if dataset is None:
+            return None
+
+        # Sampler
+        if isinstance(sampler, dict):
+            pass
+        if sampler is not None:
+            shuffle = False
+
+        dataloader = TorchDataLoader(
+            dataset,
+            batch_size=batch_size,
+            sampler=sampler,
+            num_workers=num_workers,
+            pin_memory=pin_memory,
+            shuffle=shuffle,
+            drop_last=drop_last,
+            persistent_workers=persistent_workers,
+            **kwargs,
+        )
+        sampler_cfg: dict | list | None = None
```

Comment:
it's a minor and possibly an opinionated comment, but `sampler_cfg` and `dataset_cfg` are the only variables that uses an abbreviation (cfg) as variable naming. The rest of the code does not use any abbreviation. It might be an idea to rename them to `sampler_config` and `dataset_config` in terms of consistency

Reply:
Thank you. I've renamed those words.